gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/DocumentReferenceMapper.java

```diff
@@ -0,0 +1,218 @@
+import org.hl7.v3.RCMRMT030101UK04EhrExtract;
+import org.hl7.v3.RCMRMT030101UK04NarrativeStatement;
+import org.springframework.stereotype.Service;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import uk.nhs.adaptors.pss.translator.util.DateFormatUtil;
+import uk.nhs.adaptors.pss.translator.util.EhrResourceExtractorUtil;
+import uk.nhs.adaptors.pss.translator.util.EncounterReferenceUtil;
+import uk.nhs.adaptors.pss.translator.util.ParticipantReferenceUtil;
+
+@Slf4j
+@Service
+@AllArgsConstructor
+public class DocumentReferenceMapper {
+
+    //TODO Add custodian using the reference to Organisation that is generated for sending Practice
+    //TODO Add file Size using the uncompressed/unencoded size of the document
+
+    private static final String META_PROFILE = "https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-GPC-DocumentReference-1";
+    private static final String ABSENT_ATTACHMENT = "Absent Attachment";
+    private static final String PLACEHOLDER_VALUE = "GP2GP generated placeholder. Original document not available. See notes for details";
+    private static final String INVALID_CONTENT_TYPE = "Content type was not a valid MIME type";
+
+    private CodeableConceptMapper codeableConceptMapper;
+
+    public List<DocumentReference> mapToDocumentReference(RCMRMT030101UK04EhrExtract ehrExtract, Patient patient,
+        List<Encounter> encounterList, String practiseCode) {
+        List<DocumentReference> mappedDocumentReferenceResources = new ArrayList<>();
+        var ehrCompositionList = EhrResourceExtractorUtil.extractValidDocumentReferenceEhrCompositions(ehrExtract);
+
+        ehrCompositionList.forEach(ehrComposition -> {
+            var narrativeStatements = getNarrativeStatements(ehrComposition);
+
+            narrativeStatements
+                .forEach(narrativeStatement -> {
+                    var mappedDocumentReferences = mapDocumentReference(narrativeStatement, patient, ehrExtract,
+                        encounterList, practiseCode);
+                    mappedDocumentReferenceResources.add(mappedDocumentReferences);
+                });
+        });
+
+        return mappedDocumentReferenceResources;
+    }
+
+    private DocumentReference mapDocumentReference(RCMRMT030101UK04NarrativeStatement narrativeStatement, Patient patient,
+        RCMRMT030101UK04EhrExtract ehrExtract, List<Encounter> encounterList, String practiseCode) {
+        DocumentReference documentReference = new DocumentReference();
+
+        var ehrComposition = ehrExtract.getComponent().get(0).getEhrFolder().getComponent().get(0).getEhrComposition();
+
+        var id = narrativeStatement.getReference().get(0).getReferredToExternalDocument().getId().getRoot();
+
+        documentReference.addIdentifier(buildIdentifier(id, practiseCode));
+        documentReference.getMeta().addProfile(META_PROFILE);
+        documentReference.setStatus(Enumerations.DocumentReferenceStatus.CURRENT);
```

Comment:
if you static import `Enumerations.DocumentReferenceStatus` directly all you need here is CURRENT, looks a bit cleaner